pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/table/coreimpl/MostlySingularMultimap.java

```diff
@@ -241,7 +241,7 @@ private Object appendSingle(@Nullable Object vs, V v, boolean noDuplicate) {
         public @Nullable Map<K, V> buildAsSingular() {
             consume();
             if (!isSingular) {
-                return null;
+                return Collections.emptyMap();
```

Comment:
That's ReturnEmptyCollectionRatherThanNull.
Not sure, that's correct - hm.... actually I just see `@Nullable` at the signature.... maybe we should still return null in that case?

Reply:
I've undone this change and return null again.